### PR TITLE
Fix #16404 Password generation fix after new user creation failure

### LINKED ISSUE #16404: password generation not working after failing to create new user
### Describe the bug

The "generate password" button on the /server_privileges.php?adduser=1 encountered js error after failing to create new user.
In js/functions.js, the undefined argument (this.form) was passed to the function displayPasswordGenerateButton().

### To Reproduce

1. Enable "strict password validation mode" for MariaDB 10
2. Go to phpmyadmin, click on "Add user account" on the "User accounts"
3. Create an user with weak password that does not meet the requirement of "strict password validation mode"
4. Click on "Go"
5. The user creation fails due to the "strict password validation mode" requirement
6. Click on "generate password", the password will not be generated

### Expected behavior

The "generate password" should work and allow the user to re-enter a strong password to meet the password strength requirement.

### Screenshots

Picture

### Server configuration

- Operating system: Windows 10
- Web server: nginx
- Database version: MariaDB 10 
- PHP version: PHP 7.3
- phpMyAdmin version: 4.9.6

### Client configuration

- Browser: Chrome
- Operating system: Windows 10

### PR DESCRIPTION
**Pull request by shucon**
Signed-off-by: Saksham Gupta <shucon01@gmail.com>

### Description

Please describe your pull request.

Fixes #16404 

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

**Review by williamdes — CHANGES_REQUESTED** [review]
This works fine but is invalid, the form does not go anymore to edit user form on success add

File: libraries/classes/Server/Privileges.php [hunk 1]
```diff
@@ -2772,7 +2772,7 @@ public function addUser(
                 $queries,
                 $queries_for_display,
                 $sql_query,
-                $_add_user_error,
+                $_add_user_error || $_error,
```
Comment (williamdes) [inline libraries/classes/Server/Privileges.php:2775]: do you have any idea why phpstan thinks that `Left side of || is always false.` ?

**Review by williamdes — APPROVED** [review]
Looks like a nice fix

**Comment by shucon** [comment]
@sudo-bot Deploy PR

**Comment by shucon** [comment]
@sudo-robot Deploy PR

**Pull request merged**